src/controllers/optimize.controller.ts

```diff
@@ -0,0 +1,56 @@
+import { NextFunction, Request, Response, Router } from 'express';
+import { Optimizer } from '@asyncapi/optimizer';
+
+import { validationMiddleware } from '../middlewares/validation.middleware';
+
+import { ProblemException } from '../exceptions/problem.exception';
+import { Controller } from '../interfaces';
+
+/**
+ * Controller which exposes the Optimizer functionality.
+ */
+export class OptimizeController implements Controller {
+  public basepath = '/optimize';
+
+  private async optimize(req: Request, res: Response, next: NextFunction) {
+    const { asyncapi, rules = {}, report = false } = req.body;
+
+    try {
+      const optimizer = new Optimizer(asyncapi);
+
+      const optimizerReport = await optimizer.getReport();
```

Comment:
should this run if `report` is `false`? It might be just some added milliseconds of processing time, but we could only run it if necessary on the `if` statement below:

```ts
 if (report) {
        result.report = await optimizer.getReport();
 }
```
There would be no need for an extra variable too 🙂 

wdyt @magicmatatjahu ?